include/shark/Algorithms/DirectSearch/Operators/Hypervolume/HypervolumeCalculatorND.h

```diff
@@ -0,0 +1,362 @@
+/*!
+ * 
+ *
+ * \brief       Implementation of the exact hypervolume calculation in m dimensions.
+ * 
+ * The algorithm is described in
+ * 
+ * Nicola Beume und Günter Rudolph. 
+ * Faster S-Metric Calculation by Considering Dominated Hypervolume as Klee's Measure Problem.
+ * In: B. Kovalerchuk (ed.): Proceedings of the Second IASTED Conference on Computational Intelligence (CI 2006), 
+ * pp. 231-236. ACTA Press: Anaheim, 2006. 
+ * 
+ * A specialized algorithm is used for the three-objective case:
+ *
+ * M. T. M. Emmerich and C. M. Fonseca.
+ * Computing hypervolume contributions in low dimensions: Asymptotically optimal algorithm and complexity results.
+ * In: Evolutionary Multi-Criterion Optimization (EMO) 2011.
+ * Vol. 6576 of Lecture Notes in Computer Science, pp. 121--135, Berlin: Springer, 2011.
+ *
+ *
+ * \author      T.Voss, O.Krause, T. Glasmachers
+ * \date        2014-2016
+ *
+ *
+ * \par Copyright 1995-2016 Shark Development Team
+ * 
+ * <BR><HR>
+ * This file is part of Shark.
+ * <http://image.diku.dk/shark/>
+ * 
+ * Shark is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published 
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Shark is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Shark.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMECALCULATOR_ND_H
+#define SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMECALCULATOR_ND_H
```

Comment:
How about SHARK_ALGORITHMS_DIRECTSEARCH_HYPERVOLUMECALCULATOR_MD_H ?
(replaced the ND with MD, since M usually denoted the number of objectives)